library/Businessprocess/HostNode.php

```diff
@@ -64,7 +65,20 @@ public function getUrl()
             $params['backend'] = $this->bp->getBackendName();
         }
 
-        return Url::fromPath('monitoring/host/show', $params);
+        if (Icinga::app()->isCli()) {
+            $accessible = true;
+        } else {
+            $query = $this->createBackend()->select()
+                ->from('hoststatus', array('host_name'))
+                ->where('host_name', $this->hostname);
+
+            $accessible = Monitoring::restrict($query)->fetchRow() !== false;
+        }
+
+        return Url::fromPath(
+            $accessible ? 'monitoring/host/show' : 'businessprocess/monitored-object/access-denied',
```

Comment:
Where does the `access-denied` URL lead?

Reply:
It seems I've forgotten to add some file to Git.. 😕